controllers/metal3machinetemplate_controller_test.go

```diff
@@ -70,7 +73,98 @@ var _ = Describe("Metal3MachineTemplate controller", func() {
 			Namespace: namespaceName,
 		},
 	}
+	type TestCaseM3MoM3MT struct {
+		M3Machine     *capm3.Metal3Machine
+		M3MTemplate   *capm3.Metal3MachineTemplate
+		ExpectRequest bool
+	}
+	DescribeTable("Cluster To Metal3Machines tests",
+		func(tc TestCaseM3MoM3MT) {
+			r := Metal3MachineTemplateReconciler{}
```

Comment:
I believe, we need to pass tested objects to fakeclient and call it from the Metal3MachineTemplateReconciler, i.e:
```suggestion
			objects := []client.Object{
				tc.M3Machine,
				tc.M3MTemplate,
			}
			fakeClient := fake.NewClientBuilder().WithScheme(setupScheme()).WithObjects(objects...).Build()

			r := Metal3MachineTemplateReconciler{
				Client:           fakeClient,
				Log:              logr.Discard(),
				WatchFilterValue: "",
			}
```